src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.ListView.cs

```diff
@@ -342,25 +342,21 @@ private void TasView_ColumnClick(object sender, InputRoll.ColumnClickEventArgs e
 				}
 				else if (columnName != CursorColumnName)
 				{
-					var frame = TasView.AnyRowsSelected ? TasView.FirstSelectedRowIndex : 0;
 					var buttonName = TasView.CurrentCell.Column!.Name;
 
 					if (ControllerType.BoolButtons.Contains(buttonName))
 					{
 						if (ModifierKeys != Keys.Alt)
 						{
 							// nifty taseditor logic
-							bool allPressed = true;
-							foreach (var index in TasView.SelectedRows)
+							// your TAS Editor logic failed us (it didn't account for non-contiguous `SelectedRows`) --yoshi
+							var selection = TasView.SelectedRows.ToArray(); // sorted asc, length >= 1
+							var allPressed = selection.All(index => CurrentTasMovie.BoolIsPressed(index, buttonName))
+								&& selection[selection.Length - 1] != CurrentTasMovie.FrameCount; // last movie frame can't have input, but can be selected
+							foreach (var (start, count) in selection.ChunkConsecutive())
 							{
-								if (index == CurrentTasMovie.FrameCount // last movie frame can't have input, but can be selected
-									|| !CurrentTasMovie.BoolIsPressed(index, buttonName))
-								{
-									allPressed = false;
-									break;
-								}
+								CurrentTasMovie.SetBoolStates(frame: start, count: count, buttonName, val: !allPressed);
 							}

```

Comment:
Make this produce only one undo action. Also best if it only does one invalidation since that's possible now. (I'm noticing now I didn't do changelog batching for alt column clicks...)

```cs
string set = allPressed ? "Unset" : "Set";
string batchName = $"{set} {selection.Length} frames of {buttonName} starting at {TasView.FirstSelectedRowIndex}";
CurrentTasMovie.ChangeLog.BeginNewBatch(batchName);
CurrentTasMovie.SingleInvalidation(() =>
{
	foreach (var (start, count) in selection.ChunkConsecutive())
	{
		CurrentTasMovie.SetBoolStates(frame: start, count: count, buttonName, val: !allPressed);
	}
});
CurrentTasMovie.ChangeLog.EndBatch();
```

You could even remove `ChunkConsecutive` and use the singular `SetBoolState` on each individual frame.